lib/media/media_source_engine.js

```diff
@@ -1712,6 +1740,13 @@ shaka.media.MediaSourceEngine = class {
       await sourceBufferAdded;
     } finally {
       this.reloadingMediaSource_ = false;
+
+      this.eventManager_.listenOnce(this.video_, 'canplay', () => {
```

Comment:
I think it's possible that `this.eventManager_` and `this.video_` could be null by the time the method gets here (or by the time the event manager is called), if the user loads a new asset at just the wrong time.

So this might want a call to `this.destroyer_.ensureNotDestroyed()`, before the event manager is called and inside the callback.

Reply:
Added `this.destroyer_.ensureNotDestroyed()`